github/search.go

```diff
@@ -71,6 +71,34 @@ func (s *SearchService) Repositories(ctx context.Context, query string, opt *Sea
 	return result, resp, err
 }
 
+// TopicsSearchResult represents the result of a topics search.
+type TopicsSearchResult struct {
+	Total             *int           `json:"total_count,omitempty"`
+	IncompleteResults *bool          `json:"incomplete_results,omitempty"`
+	Topics            []*TopicResult `json:"items,omitempty"`
+}
+
+type TopicResult struct {
+	Name             *string  `json:"name,omitempty"`
+	DisplayName      *string  `json:"display_name,omitempty"`
+	ShortDescription *string  `json:"short_description,omitempty"`
+	Description      *string  `json:"description,omitempty"`
+	CreatedBy        *string  `json:"created_by,omitempty"`
+	CreatedAt        *string  `json:"created_at,omitempty"`
+	UpdatedAt        *string  `json:"updated_at,omitempty"`
+	Featured         *bool    `json:"featured,omitempty"`
+	Score            *float32 `json:"score,omitempty"`
+}
+
+// Topics searches repositories via various topics.
```

Comment:
This sentence is a little bit awkward and the query parameter can contain more than just topics, it can also include qualifiers. So how about if we make this read a bit more closely to the GitHub API docs and say something like:

```go
// Topics finds topics via various criteria. Results are sorted by best match.
// Please see https://help.github.com/en/articles/searching-topics for more
// information about search qualifiers.
```

Reply:
done